analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -1703,3 +1703,33 @@ public void StringIsNullOrEmpty(string value)
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/6241
+public class Repro_6241<T>
+{
+    public void WithGenericValue()
+    {
+        HashSet<T> value = null;
+        if (value is null)
+        {
+            value.ToString();   // FN
+        }
+        else
+        {
+            value.ToString();   // Noncompliant FP
+        }
+    }
+
+    public void WithNormalValue(List<object> inputs)
```

Comment:
The parameter should not be needed.
```suggestion
    public void WithNormalValue()
```